src/components/network/network-table.js

```diff
@@ -50,10 +71,136 @@ genotet.NetworkTable.prototype.create = function(table, edges) {
+      '<"row"<"col-sm-5"i><"col-sm-7"p>>',
+    buttons: [
+      {
+        text: 'Add',
+        className: 'btn btn-default',
+        action: function(e, dt, node, config) {
+          var selectedEdges = dt.rows({selected: true}).data();
+          var additionEdges = [];
+          for (var i = 0; i < selectedEdges.length; i++) {
+            additionEdges.push({
+              id: selectedEdges[i].id,
+              source: selectedEdges[i].source,
+              target: selectedEdges[i].target,
+              weight: selectedEdges[i].originalWeight
+            });
+            edgesForTable.forEach(function(edge) {
+              if (selectedEdges[i].id == edge.id) {
+                edge.added = true;
+              }
+            });
+          }
+          this.signal('addEdges', {
+            edges: additionEdges
+          });
+
+          // refresh the row
+          dt.rows({selected: true}).invalidate();
+          // change the button status
+          dt.buttons(0).enable(false);  // the addition button
+          dt.buttons(1).enable(true);   // the removal button
+        }.bind(this),
+        enabled: false
+      },
+      {
+        text: 'Remove',
+        className: 'btn btn-default',
+        action: function(e, dt, node, config) {
+          var selectedEdges = dt.rows({selected: true}).data();
+          var removalEdges = [];
+          for (var i = 0; i < selectedEdges.length; i++) {
+            removalEdges.push({
+              id: selectedEdges[i].id,
+              source: selectedEdges[i].source,
+              target: selectedEdges[i].target,
+              weight: selectedEdges[i].originalWeight
+            });
+            edgesForTable.forEach(function(edge) {
+              if (selectedEdges[i].id == edge.id) {
+                edge.added = false;
+              }
+            });
+          }
+          this.signal('highlightEdges', {
+            edgesId: []
+          });
+          this.signal('removeEdges', {
+            edges: removalEdges
+          });
+          this.signal('hideEdge', {
+            edges: removalEdges
+          });
+          // refresh the row
+          dt.rows({selected: true}).invalidate();
+          // change the button status
+          dt.buttons(0).enable(true);   // the addition button
+          dt.buttons(1).enable(false);  // the removal button
+        }.bind(this),
+        enabled: false
+      }
+    ],
     lengthMenu: [5, 10, 20, 50],
     pageLength: 5,
     pagingType: 'full'
   });
+
+  table.on('click', function() {
+    var data = dataTable.rows({selected: true}).data();
+    if (data.length > 0) {
+      var allSame = true;
+      if (data.length > 1) {
+        for (var i = 1; i < data.length; i++) {
+          if (data[i].added != data[i - 1].added) {
+            allSame = false;
+            break;
+          }
+        }
+        this.signal('multiEdgeInfo');
+      } else {
+        if (data[0].added) {
+          this.signal('edgeInfo', {
```

Comment:
Looks like my previous version of setEdgeInfo is unnecessarily complicated. Please just pass the edge object in (network.Edge, which has source/target being the node ids). Node labels can be retrieved from the node objects, accessed by the node ids.